ott/geometry/costs.py

```diff
@@ -90,17 +90,88 @@ def tree_unflatten(cls, aux_data, children):
     return cls(*children)
 
 
+@jax.tree_util.register_pytree_node_class
+class RBFCost(CostFn):
+  """A radial-basis function cost class for translation invariant costs.
+
+  Such costs are defined as
+
+  c(x,y) = h(z), where z := x-y.
+
+  where h is a function strictly convex (or concave) function mapping vectors
+  to real-values.
+
+  For completeness (and differentiation using the Brenier theorem), the user
+  is also supposed to provide the Legendre transform of `h`, whose gradient (the
+  inverse of the gradient of `h`) will be used to form a Brenier map.
+  """
+
+  def h(self, z: jnp.ndarray) -> float:
```

Comment:
Unless I'm mistaken, there is no ABC mechanism involved here to prevent direct calls to `RBFCost` or incomplete user-provided subclasses. This is totally fine, but maybe it would be nicer to remind the user that she had to provide `h` and `h_legendre`? I think it would yield a nicer error message in case she forgot. Currently, she would get a message like "None is not callable". 

Maybe something along the lines of

```python
    raise NotImplementedError("The function `h` must be provided when implementing `RBFCost`.")
```

and similarly for `h_legendre`?

Reply:
thanks a lot for this great remark! 

I think what you said makes total sense for `h`. Enforcing this with `@abc.abstractmethod` makes actually a lot of sense.

For `h_legendre` the user can define a `RBFCost` but I can envision cases where the user uses a function `h` that's not necessarily convex (e.g. square root of norm), or for which `h_legendre` is not known. In that case, the user can still run any OT solver (Sinkhorn / GW etc), but cannot use it to turn it into an entropic map,  which would be fine for now. In other words, `h_legendre` is really only needed when turning the output of Sinkhorn into a `dual_potentials`.